content/blog/2021/containerqueries.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Getting Started with Container Queries
+sub: Now is the time to begin experimenting with a long requested layout tool. 
+author: davidh
+date: 2021-04-02
+image:
+  src: blog/2021/widequote.jpg
+  alt: blockquote with magenta background and polygonal clip-path
+tags:
+  - CSS
+  - CSSWG
+  - Article
+  - Container Queries
+summary: |
+  The `@container` query, that elusive feature developers have been requesting and proposing for years, has finally made its debut in browsers. Well, sort of. What are **container queries** exactly and how do they work?
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+  In the latest version of [Chrome Canary](https://www.google.com/chrome/canary), the most recent `@container` query proprosal, developed by OddBird's own [Miriam Suzanne](https://www.oddbird.net/authors/miriam/), is available for use behind an experimental flag. Miriam's [proposal](https://github.com/w3c/csswg-drafts/issues/5796) builds upon the ideas previously proposed by others, in particular David Baron, who wrote the [original draft](https://github.com/dbaron/container-queries-implementability). 
+
+
+ ## Background
+
+`@Media` queries sparked a responsive design revolution by allowing authors to change the style of elements based on the size of the entire viewport. Up until now, what could not be done was changing the style of an element based on the size of its parent element. 
+
+## Syntax and an example
+
+ The proposed `@container` query syntax feels a lot like writing a `@media` query, but a main difference is that a `@container` query has to be implemented in two parts.
+
+In this demo (which only works in the latest version of [Chrome Canary](https://www.google.com/chrome/canary)), a `blockquote` is styled differently depending on the size of its container. 
+
+{{ embed.codepen(
+  id='YzpywrZ',
+  title='Container Queries with Block Quotes',
+  user='dvdherron',
+  height=600
+) }}
+
+### Define a containment context
+
+To achieve this, first define a containment context. This lets the browser know which container to query against later. 
+
+```scss
+article,
+section {
+  contain: layout inline-size;
+}
+```
+
+The `contain` property is part of the existing [CSS Containment Module](https://drafts.csswg.org/css-contain/). The `layout` value activates [layout contaiment](https://drafts.csswg.org/css-contain/#valdef-contain-layout) on the container, which ensures that "nothing outside can affect its internal layout, and vice versa." `Size` also [currently exists](https://drafts.csswg.org/css-contain/#size-containment) as a value for the `contain` property, and enables laying out a containment box without accounting for its descendents in either the inline or block axes. 
+
+```scss
+article,
+section {
+  contain: layout size;
+}
+```
```

Comment:
switch the order of these two code samples, since you talk about `size` in the paragraph above, and `inline-size` below?

Reply:
fixed